### ISSUE TITLE
LLVM Source-Based coverage incorrect reports missing coverage on closing brace on switch statements

### ISSUE THREAD

**Issue by carlosgalvezp** [llvm/llvm-project#64956]
Consider the following code:

```cpp
#include <iostream>

enum E
{
    a,
    b,
    c
};

void foo(E e)
{
    switch (e)
    {
        case a:
        {
            std::cout << "a\n";
            break;
        }  // <<<<<<<<<<< Missing coverage
        case b:
        {
            std::cout << "b\n";
            break;
        }  // <<<<<<<<<<< Missing coverage
        case c:
        {
            std::cout << "c\n";
            break;
        }  // <<<<<<<<<<< Missing coverage
    }
}

int main()
{
    foo(a);
    foo(b);
    foo(c);
}
```

Running source-based coverage as per the docs leads to 88% coverage instead of 100% coverage, because the braces after the "break" are considered as "executable" but "not-executed".

Is this intended? If not, is it possible to configure LLVM to ignore braces when computing coverage statistics?

Thanks!

**Comment by cbachhuber** [comment]
I'm running into the same problem. One workaround is to remove the braces around the case bodies.

Obviously, this is just a workaround and the issue should be fixed.

**Comment by justincady** [comment]
This is still a problem today. Is this something that can be addressed, or could someone knowledgable in this area provide guidance for another to pick this up?

cc @chapuni @evodius96 

Thanks!

**Comment by evodius96** [comment]
I'm not aware of the history for this behavior, but it also looks incorrect to me. Thank you for highlighting this. Would it be worth a question on Discourse to draw more attention to it?

I could try to dig into it a bit, but I'm dealing with higher priority issues over the next few weeks.

**Comment by soerenkoehler** [comment]
> I'm not aware of the history for this behavior, but it also looks incorrect to me. Thank you for highlighting this. Would it be worth a question on Discourse to draw more attention to it?
> 
> I could try to dig into it a bit, but I'm dealing with higher priority issues over the next few weeks.

Would much appreciate it since this behaviour just drove me crazy.

In my case it was in an if-else-construct, where llvm reported the final brace of the else block as uncovered. Since I'm using rust, at first I assumed the rust compiler would do something strange like inserting extra checks which the llvm coverage tools may detect as uncovered by my tests. But whatever code reorganisation I tried, it always reported the final brace. 🤷‍♂️

![Image](https://github.com/user-attachments/assets/400794c6-69e7-4c74-8805-0e6cbef61859)
